src/Pim/Component/Catalog/Model/ProductUniqueData.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Pim\Component\Catalog\Model;
+
+/**
+ * @author    Julien Janvier <julien.janvier@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProductUniqueData implements ProductUniqueDataInterface
+{
+    /** @var int */
+    protected $id;
+
+    /** @var ProductInterface */
+    protected $product;
+
+    /** @var ProductValueInterface */
+    protected $value;
```

Comment:
@grena the ProductValue is here used to set correctly the $rawData.
With the product value, I have the attribute and the data. That's why I use the product value in the constructor of the ProductUniqueData, and it avoid have a setRawData, which I find totally ugly.

Indeed, only $rawData is stored, but we can use whichever object we want inside an entity.